modules/system/assets/js/framework.js

```diff
@@ -306,7 +319,9 @@ if (window.jQuery.request !== undefined) {
         context.error = requestOptions.error
         context.complete = requestOptions.complete
         requestOptions = $.extend(requestOptions, options)
-        requestOptions.data = data.join('&')
+        requestOptions.data = sendFiles ? formdata : data
```

Comment:
As the data is now an array of objects with name and value properties join('&') is no longer needed as $.ajax accepts data in this format.